p2p/protocol/identify/id.go

```diff
@@ -369,10 +372,13 @@ func (ids *IDService) identifyConn(c network.Conn, signal chan struct{}) {
 		s.Reset()
 		return
 	}
-	ids.handleIdentifyResponse(s)
+
+	err = ids.handleIdentifyResponse(s)
 }
 
 func (ids *IDService) sendIdentifyResp(s network.Stream) {
+	s.SetDeadline(time.Now().Add(StreamReadTimeout))
```

Comment:
nit: using the read timeout here is a bit strange, this mostly applies to writes.

Reply:
I agree, had put in just in case. Have removed it.